pkg/event/target/mqtt.go

```diff
@@ -96,6 +96,12 @@ func (target *MQTTTarget) Send(eventData event.Event) error {
 
 	token := target.client.Publish(target.args.Topic, target.args.QoS, false, string(data))
 
+	// If the client is in disconnected or reconnect mode, The events are backed up
+	// in `/root/.minio/mqtt/` and are retried when the connection is back
+	if !target.client.IsConnectionOpen() {
```

Comment:
this check needs to be above `target.client.Publish()`?

Reply:
@balamurugana  No, the `Publish()` writes the events to the store directory through `store.put()` calls. If the connections is active/open, It returns a token to track delivery of the message to the broker. The token is not returned in this case ( if the connection is not active / not open )

Reply:
If so, you would need to check token validity not the connection.

Reply:
*Sorry the token is returned all the time. We have to wait until the broker sends ACK back ( `token.wait()` ). Which will not happen if the broker is disconnected. ( the token.wait() will return true until the broker acknowledges)

And the token interface has
```
Which does not expose any information about the connections. I think `target.client.IsConnectionOpen()` is only way to know about the connection.
type Token interface {
	Wait() bool
	WaitTimeout(time.Duration) bool
	Error() error
}
```